collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java

```diff
@@ -263,16 +267,37 @@ private void insertRow(TreeElement element, String parentKey, String key, File i
                 sheetCells = getSheetCells(sheetTitle); // read sheet cells again to update
             }
 
-            HashMap<String, String> answers = getAnswers(element, instanceFile, parentKey, key);
+            List<Object> valuesList = prepareListOfValues(sheetCells.get(0), columnTitles, answers);
+
             if (shouldRowBeInserted(answers)) {
                 sheetsHelper.insertRow(spreadsheet.getSpreadsheetId(), sheetTitle,
-                        new ValueRange().setValues(Collections.singletonList(prepareListOfValues(sheetCells.get(0), columnTitles, answers))));
+                        new ValueRange().setValues(Collections.singletonList(valuesList)));
             }
         } catch (IOException e) {
             throw new UploadException(e);
         }
     }
 
+    /**
+     * addExtraTitles
+     *
+     * Adds extra column titles that could have been added to the answers Map. This is primarily
+     * to add columns for Accuracy and Altitude.
+     *
+     *
+     * @param columnTitles List of current column titles
+     * @param answers Map of answers
+     * @return column titles
+     */
+    private List<Object> addExtraTitles(@NonNull List<Object> columnTitles, @NonNull Map<String, String> answers) {
```

Comment:
Another approach that might be more targeted and match the existing approach of getting column titles from the form rather than the results would be to modify `getColumnTitles` with something like the following in the loop over children after adding the child:
```
if (child.getDataType() == org.javarosa.core.model.Constants.DATATYPE_GEOPOINT) {
    columnTitles.add(getElementTitle(child) + "-accuracy");
    columnTitles.add(getElementTitle(child) + "-altitude";
}